### PR TITLE
Sam tov memory scaling function

### PR DESCRIPTION
**Pull request by SamTov**
Added memory fraction as a configuration parameter

#### Summary of additions and changes

* Added  used memory fraction as a configuration parameter
* Updated memory manager to use this

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.x'
+          python-version: '3.9'
```
Comment (PythonFZ) [inline .github/workflows/lint.yaml:49]: Should be fixed on all linting?

File: mdsuite/utils/config.py [hunk 2]
```diff
@@ -36,11 +36,15 @@ class Config:
     bokeh_sizing_mode: str
         The way bokeh scales plots.
         see bokeh / sizing_mode for more information
+    jupyter : bool
+            If true, jupyter is being used.
```
Comment (PythonFZ) [inline mdsuite/utils/config.py:40]: What does this mean?

Comment (SamTov) [reply mdsuite/utils/config.py:40]: If true, the environment is jupyter.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (SamTov) [reply .github/workflows/lint.yaml:49]: Perhaps, we should probably open an issue / PR and discuss it as sometimes it is nice for it to be latest but a fixed one for linting is probably okay.

**Review by PythonFZ — APPROVED** [review]

**Pull request merged**